wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/OptionElement.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2013 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema.internal.parser
+
+import com.squareup.wire.schema.internal.Util
+import com.squareup.wire.schema.internal.Util.appendIndented
+
+data class OptionElement(
+  val name: String,
+  val kind: Kind,
+  val value: Any,
+  private val isParenthesized: Boolean
+) {
+  enum class Kind {
+    STRING,
+    BOOLEAN,
+    NUMBER,
+    ENUM,
+    MAP,
+    LIST,
+    OPTION
+  }
+
+  fun toSchema(): String = buildString {
+    when (kind) {
+      Kind.STRING -> return """${formatName()} = "$value""""
+      Kind.BOOLEAN,
+      Kind.NUMBER,
+      Kind.ENUM -> return "${formatName()} = $value"
+      Kind.OPTION -> {
+        // Treat nested options as non-parenthesized always, prevents double parentheses.
+        val optionValue = (value as OptionElement).run {
+          OptionElement.create(name, kind, value)
+        }
```

Comment:
I think this is just `OptionElement.create(name, kind, value as OptionElement)`

Reply:
It's more like `val optionValue = (value as OptionElement).copy()`.  Updated.